### ISSUE TITLE
Duplicate detection of same object with multiple classes

### ISSUE THREAD

**Issue by saumilsdk** [IDEA-Research/Grounded-Segment-Anything#279]
Model is unable to handle to detect right class for the object and detects multiple classes for the same object. Issue is seen by @utkarshtheone

![image](https://github.com/IDEA-Research/Grounded-Segment-Anything/assets/8816702/cee89ea7-8aca-40d1-a68d-db5e7d40706c)

**Comment by rentainhe** [comment]
Yes, it's the issue of our groundingdino post process, we will combine the words with high similarity, so there's some duplicate predictions, we will modify the post process in the future release

**Comment by saumilsdk** [comment]
Thanks @rentainhe for the update. We also see confusion with combined words as we don't know when the words are getting combined with high similarity and when it is single item. e.g here "autorikshaw car" are two objects but "traffic signal" is one whole class.
